File: jdroid-gradle-commons-plugin/src/main/java/com/jdroid/gradle/commons/versioning/IncrementMajorVersionTask.java
package com.jdroid.gradle.commons.versioning;

public class IncrementMajorVersionTask extends AbstractIncrementVersionTask {
	
	public IncrementMajorVersionTask() {
		setDescription("Increments the major version (X.Y.Z) -> (X+1.0.0)");
	}

	@Override
	protected void incrementVersion(Version version) {
		version.incrementMajor();
	}

}

File: jdroid-gradle-commons-plugin/src/main/groovy/com/jdroid/gradle/commons/versioning/IncrementMajorVersionTask.kt
package com.jdroid.gradle.commons.versioning

open class IncrementMajorVersionTask : AbstractIncrementVersionTask() {

    init {
        description = "Increments the major version (X.Y.Z) -> (X+1.0.0)"
    }

    override fun incrementVersion(version: Version) {
        version.incrementMajor()
    }
}
